integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/AcknowledgementsTestCase.java

```diff
@@ -0,0 +1,308 @@
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+
+
+/**
+ * This class includes test cases to test auto acknowledgements modes for queues
+ */
+public class AcknowledgementsTestCase {
+
+    /**
+     * The amount of messages to be sent.
+     */
+    private static final int SEND_COUNT = 1000;
+
+    /**
+     * The amount of messages to be expected.
+     */
+    private static final int EXPECTED_COUNT = SEND_COUNT;
+
+    /**
+     * The logger used in logging information, warnings, errors and etc.
+     */
+    private static Logger log = Logger.getLogger(QueueTestCase.class);
+
+    /**
+     * Initiate new server manager instance.
+     */
+    private ServerManager serverManager = new ServerManager();
+
+    /**
+     * initiate configuration reader instance
+     */
+    private ConfigurationReader configurationReader = new ConfigurationReader();
+
+    /**
+     * create config map
+     */
+    private Map<String, String> clientConfigPropertiesMap;
+
+    /**
+     * Initialise test environment.
+     */
+    @BeforeClass
+    public void init() {
+
+        String archiveFilePath = System.getProperty("carbon.zip");
+
+        // Create file instance for given path.
+        File distributionArchive = new File(archiveFilePath);
+
+        // Verify if given archive path is a file and not a directory before proceed.
+        if (distributionArchive.exists() && !distributionArchive.isDirectory()) {
+            try {
+                String tempCarbonHome = serverManager.setupServerHome(archiveFilePath);
+
+                // load client configs to map
+                clientConfigPropertiesMap = configurationReader.readClientConfigProperties();
+
+                // Start Enterprise Integrator broker instance
+                serverManager.startServer(tempCarbonHome);
+            } catch (IOException e) {
+                log.error("IO exception occured when trying to initialize server environment", e);
+            }
+        }
+    }
+
+    /**
+     * In this method we just test a sender and receiver with acknowledgements.
+     * 1. Start a queue receiver in auto acknowledge mode.
+     * 2. Publisher sends {@link #SEND_COUNT} amount of messages.
+     * 3. Receiver receives {@link #EXPECTED_COUNT}
+     * 4. Check whether all messages received.
+     *
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test
+    public void autoAcknowledgementsTestCase()
+            throws JMSException, NamingException, IOException, InterruptedException {
+
+        // Creating a JMS consumer
+        QueueReceiver queueReceiver = new QueueReceiver("autoAckTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
```

Comment:
Thanks unwanted sleeps used after subscriptions are removed in 06072de2be2dab2bc10d61cad43337340f61fe71